crates/engine/tree/src/tree/mod.rs

```diff
@@ -91,6 +91,11 @@ pub mod state;
 /// backfill this gap.
 pub(crate) const MIN_BLOCKS_FOR_PIPELINE_RUN: u64 = EPOCH_SLOTS;
 
+/// The max number of blocks to persist in batch.
+/// Memory is released after blocks are persisted in a batch, so limiting the batch size
+/// prevents memory accumulation and ensures timely cleanup in high-throughput scenarios.
+const MAX_BLOCKS_TO_PERSIST: u64 = 8;
```

Comment:
is there a reason why we are using 8 here? maybe a comment about the config would be good